src/main/java/com/example/animalshelter/repository/AnimalRepositoryListImpl.java

```diff
@@ -19,15 +19,19 @@ public boolean add(Animal animal) {
     }
 
     @Override
-    public boolean delete(Integer id) throws DeleteAnimalException {
-        return animals.remove(getAnimalWithId(id));
+    public Animal delete(Integer id) throws AnimalNotFoundException {
+         Animal animalToBeDeleted = getAnimalWithId(id);
+         if(animals.remove(animalToBeDeleted)) {
+             return animalToBeDeleted;
+         }
+         throw new AnimalNotFoundException(ANIMAL_NOT_FOUND_MSG);
```

Comment:
I think in your case it will never happen as we are operating on the list and checking the presence of the element before.
If you really need to check it then the exception also is a different one as you do not know the reason for failure here, do you?

Reply:
Yes you are right, I will remove this.